File: answer.py
input=list('The domestic dog is a domesticated form of wolf')

str1=''.join(input).casefold().split()

str2=''.join(input).casefold().split()
str2.sort()

for i in range(len(str2)):
    for j in range(i+1,len(str2)):
        if (str2[i][0]==str2[j][0]):
            if (str1.index(str2[i])>str1.index(str2[j])):
                temp =str2[i]
                str2[i]=str2[j]
                str2[j]=temp
answer = ' '.join(str2)
print(answer)

File: answer.js
var str = "the domestic dog is a domesticated form of wolf"; //"this is an impossible task";
var ar = Array.from(str);
console.log(ar);
 
var spaces = [];
 
for (var i = 0; i < ar.length; i++) {
  if (ar[i] == " ") {
    console.log(i);
    spaces.push(i);
  }
}
console.log(spaces);
 
var alpha = "abcdefghijklmnopqrstuvwxyz";
var alpha_ar = Array.from(alpha);
console.log(alpha_ar);
 
var pos = [0];
for (var i = 0; i < spaces.length; i++) {
  pos.push(spaces[i] + 1);
}
console.log(pos);
 
var result = "";
for (var i = 0; i < alpha_ar.length; i++) {
  for (var j = 0; j < pos.length; j++) {
    if (alpha_ar[i] == ar[pos[j]]) {
      for (var k = pos[j]; k < ar.length; k++) {
        if (ar[k] != " ") {
          result = result + ar[k];
        } else {
          break;
        }
      }
      result += " ";
    }
  }
}
console.log(result);